src/modules/75-ce/components/COInstanceSelector/COInstanceSelector.tsx

```diff
@@ -324,12 +358,55 @@ const InstancesFilter: React.FC<InstancesFilterProps> = ({
     handleGcpZonesFilterUpdate()
   }, [zonesData])
 
+  useEffect(() => {
+    if (tagsData?.response?.length) {
+      const keys = tagsData.response.map(tag => ({ label: tag.key as string, value: tag.key as string }))
+      setTagKeys(keys)
+    }
+  }, [tagsData])
+
+  useEffect(() => {
+    if (selectedTagPair?.key) {
+      const selectedTag = tagsData?.response?.find(tag => tag.key === selectedTagPair.key)
+      setTagValues(
+        _defaultTo(
+          selectedTag?.values?.map(v => ({ label: v, value: v })),
+          []
+        )
+      )
+    }
+  }, [selectedTagPair?.key])
+
+  useEffect(() => {
+    if (selectedTagPair?.key && selectedTagPair.value) {
+      onAwsFiltersChangeCallback({ region: selectedRegion, tags: selectedTagPair }, false)
+    }
+  }, [selectedTagPair])
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has missing dependencies: 'onAwsFiltersChangeCallback' and 'selectedRegion'. Either include them or remove the dependency array. If 'onAwsFiltersChangeCallback' changes too often, find the parent component that defines it and wrap that definition in useCallback.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9255099)
```suggestion
  }, [onAwsFiltersChangeCallback, selectedRegion, selectedTagPair])
```